plugins/plugin.go

```diff
@@ -141,32 +145,38 @@ func (p *Plugin) StartPlugin() error {
 // Metrics polls the DC/OS components and returns a slice of
 // producers.MetricsMessage.
 func (p *Plugin) Metrics() ([]producers.MetricsMessage, error) {
-	if len(p.AuthToken) == 0 {
-		return nil, errors.New("Auth token must be set, use --auth-token <token>")
-	}
 
 	metricsMessages := []producers.MetricsMessage{}
+
+	// The first time Metrics() is called, the plugin client
+	// should be initialised
+	if p.Client == nil {
```

Comment:
why not make `p.Client` as a required argument to [`New` constructor?](https://github.com/dcos/dcos-metrics/blob/e4ae35b8e2bd12d208eb9caeea8c0315fdc72437/plugins/plugin.go#L60) you will be 100% sure the http client is initialized that way